### PR TITLE
map component View methods - 2499

### PR DESCRIPTION
**Pull request by hdamr**
Adds IT for view methods.
Adds some comment for View class.
Do an extra check for setting the center not to be null.

### CODE REVIEW

File: vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/configuration/View.java [hunk 1]
```diff
@@ -28,33 +42,62 @@ public View() {
...
+    /**
+     * Gets center coordinates of the map in epsg:3857 format
+     * @return center of the map in epsg:3857 format
+     */
     public Coordinate getCenter() {
         return center;
     }
 
+    /**
+     * Sets the center of the map
```
Comment (sissbruecker) [inline vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/configuration/View.java:62]: The coordinates specify the center of the viewport, not of the map. The center of the map is a different concept, so it's important to not get these mixed up.

Similarly the other JavaDoc should refer to the viewport, as well.

File: vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/configuration/View.java [hunk 2]
```diff
@@ -16,10 +16,24 @@
  * #L%
  */
 
+import java.util.Objects;
+
+/**
+ * View is responsible for centering of the map, rotation or zoom level
```
Comment (sissbruecker) [inline vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/configuration/View.java:22]: We're not really modifying the map here (centering or rotating), but the viewport for a map.

Also the comment kind of implies an assumption that the view only consists of these 3 things. But in fact there are a lot more. I think we should keep it more general.

**Comment by sissbruecker** [comment]
JavaDoc for [`MapBase.getView`](https://github.com/vaadin/flow-components/blob/feat/map/vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/MapBase.java#L49) is missing.

File: vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/configuration/View.java (hunk 2 above)
Comment (hdamr) [reply vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/configuration/View.java]: Done.

File: vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/configuration/View.java (hunk 1 above)
Comment (hdamr) [reply vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/configuration/View.java]: Done.

File: vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/configuration/View.java [hunk 3]
```diff
@@ -28,33 +42,62 @@ public View() {
         this.zoom = 0;
     }
 
+    /**
+     * This method is used for internal synchronization of map configuration.
+     */
```
[the remark this answers is not in the document]
Comment (hdamr) [reply vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/configuration/View.java]: Done.

File: vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/configuration/View.java [hunk 4]
```diff
@@ -28,33 +42,62 @@ public View() {
...
     @Override
     public String getType() {
         return Constants.OL_VIEW;
     }
 
+    /**
+     * Gets center coordinates of the map in epsg:3857 format
+     * @return center of the map in epsg:3857 format
```
[the remark this answers is not in the document]
Comment (hdamr) [reply vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/configuration/View.java]: Done.

File: vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/configuration/View.java (hunk 1 above)
Comment (hdamr) [reply vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/configuration/View.java]: Done.

**Review by sissbruecker — APPROVED** [review]

**Pull request merged**